File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProcess/IfcEvent/IfcEvent.py
class IfcEvent(IfcProcess):
    def __init__(self):
        self.PredefinedType: IfcEventTypeEnum = None
        self.EventTriggerType: IfcEventTriggerTypeEnum = None
        self.UserDefinedEventTriggerType: IfcLabel = None
        self.EventOccurenceTime: IfcEventTime = None

    # === WHERE CLAUSES ===
    # CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcEventTypeEnum.USERDEFINED) OR ((PredefinedType = IfcEventTypeEnum.USERDEFINED) AND EXISTS(SELF\IfcObject.ObjectType))
    # CorrectTypeAssigned : NOT(EXISTS(EventTriggerType)) OR (EventTriggerType <> IfcEventTriggerTypeEnum.USERDEFINED) OR ((EventTriggerType = IfcEventTriggerTypeEnum.USERDEFINED) AND EXISTS(UserDefinedEventTriggerType))

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProcess/IfcEvent/IfcEvent.cs
public class IfcEvent : IfcProcess
{
    public IfcEventTypeEnum PredefinedType { get; set; }
    public IfcEventTriggerTypeEnum EventTriggerType { get; set; }
    public IfcLabel UserDefinedEventTriggerType { get; set; }
    public IfcEventTime EventOccurenceTime { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcEventTypeEnum.USERDEFINED) OR ((PredefinedType = IfcEventTypeEnum.USERDEFINED) AND EXISTS(SELF\IfcObject.ObjectType))
    // CorrectTypeAssigned : NOT(EXISTS(EventTriggerType)) OR (EventTriggerType <> IfcEventTriggerTypeEnum.USERDEFINED) OR ((EventTriggerType = IfcEventTriggerTypeEnum.USERDEFINED) AND EXISTS(UserDefinedEventTriggerType))
}
